src/components/VuePageTransition.vue

```diff
@@ -1,43 +1,46 @@
 <template>
   <div>
     <transition
-      :name="transition"
-      :mode="mode"
+      :name="model.name"
+      :mode="model.mode"
     >
       <slot></slot>
     </transition>
-    <div class="overlay-top"></div>
-    <div class="overlay-right"></div>
-    <div class="overlay-bottom"></div>
-    <div class="overlay-left"></div>
+    <div class="overlay overlay-top" :style="model.overlay"></div>
+    <div class="overlay overlay-right" :style="model.overlay"></div>
+    <div class="overlay overlay-bottom" :style="model.overlay"></div>
+    <div class="overlay overlay-left" :style="model.overlay"></div>
   </div>
 </template>
 
 <script>
 export default {
   name: 'vue-page-transition',
-  props: ['name'],
   data () {
     return {
-      transition: 'fade',
-      mode: 'out-in',
+      model: {
+        mode: 'out-in',
+        name: null,
+        overlay: {
+          background: '#1867c0'
+        }
+      }
     }
   },
+  props: ['mode', 'name', 'overlay'],
   created () {
     this.$router.beforeEach((to, from, next) => {
-      this.transition = to.meta.transition
-        ? to.meta.transition
-        : this.$props.name
-
+      this.model.mode = to.meta.transition.mode || this.$props.mode || this.model.mode
+      this.model.name = to.meta.transition.name || this.$props.name || this.model.name
```

Comment:
This is why the default value should be null. (It can be avoid by creating a "none" transition)